### PR TITLE
Closes #48 | Added Modal component

### LINKED ISSUE #48: Component: Modal
Create a modal component that can be used throughout the site. See this [article](https://alligator.io/vuejs/vue-modal-component/) for an example of how to make an accessible modal in Vue that uses slots for reusable content.

- [ ] Title
- [ ] Close Button
- [ ] Content

### PR DESCRIPTION
**Pull request by jdromero88**
Closes #48 
This is the default modal with the default css styles from the [article](https://alligator.io/vuejs/vue-modal-component/).

### CODE REVIEW

**Review by jnschrag — CHANGES_REQUESTED** [review]
Thank you, José! Appreciate you taking the initiative on this. Can you do a couple of things for me?

- I like the fade transition from the article. Can you add that here?
- Our modal is going to look like this [Figma mockup](https://www.figma.com/file/XzGzlXpNmeB3PzTNfzk4ro/Satellite-Dashboard%2C-handoff?node-id=84%3A91). Can you take a first pass at updating the background & font colors? The footer will be that darker section at the bottom. Please don't worry about any of the content - I will handle that separately.
- Let's leverage Sass nesting for this. So your CSS should look like this. Don't forget about using BEM for naming!

```
.modal {
  &__backdrop {
    // styles here
  }

  &__footer {
    // styles here
  }
}
```

File: assets/css/components/modal.scss [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    padding: rem(16) rem(32);
+    color: $color-white-150;
+    text-transform: uppercase;
+    background-color: $color-slate-400;
+    @extend %font-ui-text-base-good;
+  }
+
+  &__body {
+    position: relative;
+    padding: 20px 10px;
```
Comment (jnschrag) [inline assets/css/components/modal.scss:56]: Can you turn these into rems, please?

**Review by jnschrag — CHANGES_REQUESTED** [review]
This looks good to me! Can you fix the one thing I noted about changing px to rems? and then delete the test.vue page? Thank you!

**Comment by jdromero88** [comment]
Hola Jacque, I converted to draft because I refactor the ```modal.scss```  & I left the test so you can check it out. After you review I can deleted the test and push it again & update the PR to remove the draft mode.
Now everything is nested under ```.modal```

**Review by jnschrag — APPROVED** [review]
This looks great to me. Thanks for cleaning it up! Once you change this back to published & remove the test page, I'll merge it in. :) Can you keep a copy of the code from `test.vue` though? You can save it as a comment in response to this PR. Then when we start work on the actual modal component, we have it as a reference. :D

**Comment by jdromero88** [comment]
This code is is for testing the modal and should be located in ```pages/test.vue```
```
<template>
  <div class="test">
    <Button class="btn--controls" :on-click="showModal">
      Open Modal!
    </Button>
    <Modal v-show="isModalVisible" @close="closeModal">
      <template v-slot:header>
        Lorem Ipsum
        <Button
          type="button"
          class="modal__btn-close-icon"
          aria-label="Close modal"
          :on-click="closeModal"
        >
          <Icon id="close-small" name="close-small" />
        </Button>
      </template>
      <template v-slot:body>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam commodo
          consequat lectus, a fringilla sem dictum gravida. Suspendisse semper
          volutpat ipsum et luctus. Morbi a iaculis lectus. Proin sit amet
          fermentum est, nec varius ante.
        </p>
      </template>
      <template v-slot:footer>
        <div class="modal__footer-left">
          <Button type="button" class="modal__btn-footer" aria-label="Print">
            <Icon id="printer" name="printer" />
            Printer
          </Button>
        </div>
        <div class="modal__footer-right">
          <Button
            type="button"
            class="modal__btn-footer"
            aria-label="Close modal"
            :on-click="closeModal"
          >
            CSV
          </Button>
          <Button
            type="button"
            class="modal__btn-footer"
            aria-label="Close modal"
            :on-click="closeModal"
          >
            JPG
          </Button>
          <Button
            type="button"
            class="modal__btn-footer"
            aria-label="Close modal"
            :on-click="closeModal"
          >
            SVG
          </Button>
        </div>
      </template>
    </Modal>
  </div>
</template>

<script>
import Icon from '~/components/global/Icon.vue'
import Button from '~/components/global/Button.vue'
import Modal from '~/components/global/Modal'

export default {
  name: 'Test',
  components: {
    Icon,
    Button,
    Modal
  },
  data() {
    return {
      isModalVisible: false
    }
  },
  methods: {
    showModal() {
      this.isModalVisible = true
    },
    closeModal() {
      this.isModalVisible = false
    }
  }
}
</script>

<style>
.test {
  display: flex;
  justify-content: center;
}
</style>
```

**Review by jnschrag — APPROVED** [review]

**Pull request merged**